pkg/scheduler/controller/shoot/scheduler_control.go

```diff
@@ -306,51 +354,83 @@ func determineCandidatesOfSameProvider(seedList []*gardencorev1beta1.Seed, shoot
+		if seed.Spec.Provider.Type == shoot.Spec.Provider.Type && seed.Spec.Provider.Region == shoot.Spec.Region {
 			candidates = append(candidates, seed)
 		}
 	}
 	return candidates
 }
 
 func determineCandidatesWithMinimalDistanceStrategy(seeds []*gardencorev1beta1.Seed, shoot *gardencorev1beta1.Shoot) []*gardencorev1beta1.Seed {
-	if sameRegionCandidates := determineCandidatesWithSameRegionStrategy(seeds, shoot); sameRegionCandidates != nil {
-		return sameRegionCandidates
-	}
-
 	var (
-		maxMatchingCharacters int
-		shootRegion           = shoot.Spec.Region
-		candidates            []*gardencorev1beta1.Seed
+		minDistance   = 1000
+		shootRegion   = shoot.Spec.Region
+		shootProvider = shoot.Spec.Provider.Type
+		candidates    []*gardencorev1beta1.Seed
 	)
 
 	for _, seed := range seeds {
-		if seed.DeletionTimestamp == nil && seed.Spec.Provider.Type == shoot.Spec.Provider.Type && seed.Spec.Settings.Scheduling.Visible && common.VerifySeedReadiness(seed) {
-			seedRegion := seed.Spec.Provider.Region
-			var matchingCharacters int
-			for i := 0; i <= len(shootRegion); i++ {
-				prefix := shootRegion[:i]
-				if strings.HasPrefix(seedRegion, prefix) {
-					matchingCharacters = len(prefix)
-					continue
-				}
-			}
+		seedRegion := seed.Spec.Provider.Region
+		dist := distance(seedRegion, shootRegion)
 
-			// append
-			if matchingCharacters == maxMatchingCharacters {
-				candidates = append(candidates, seed)
-				continue
-			}
-			// replace
-			if matchingCharacters > maxMatchingCharacters {
-				maxMatchingCharacters = matchingCharacters
-				candidates = []*gardencorev1beta1.Seed{seed}
-			}
+		if shootProvider != seed.Spec.Provider.Type {
+			dist = dist + 2
+		}
+		// append
+		if dist == minDistance {
+			candidates = append(candidates, seed)
+			continue
+		}
+		// replace
+		if dist < minDistance {
+			minDistance = dist
+			candidates = []*gardencorev1beta1.Seed{seed}
 		}
 	}
 	return candidates
 }
 
+var orientations = []string{"north", "south", "east", "west", "central"}
+
+// orientation extracts an orientation relative to a base from a region name
+func orientation(name string) (normalized string, orientation string) {
+	for _, o := range orientations {
+		if i := strings.Index(name, o); i >= 0 {
+			orientation = o
+			normalized = name[:i] + ":" + name[i+len(o):]
+			return
+		}
+	}
+	return name, ""
+}
+
+// distance calculates a formal distance between two region names observing
+// some usual orientation keywords. It is based on the levenshtein distance
+// of the regions base names plus the difference based on the orientation.
+// regions with the same base but different orientations are basically nearer
+// to each other than two completely unrelated regions.
+func distance(seed, shoot string) int {
+	d, _ := _distance(seed, shoot)
+	return d
+}
+
+func _distance(seed, shoot string) (int, int) {
+	seed_base, seed_orient := orientation(seed)
```

Comment:
```suggestion
	seedBase, seedOrientation := orientation(seed)
```

Ref https://golang.org/doc/effective_go.html#mixed-caps